daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/JsonApiIt.scala

```diff
@@ -270,7 +271,16 @@ final class JsonApiIt
       )
       .toMap
     val participantParams = Participants(Some(defaultParticipant), participantMap, partyMap)
-    Runner.jsonClients(participantParams, envIface)
+    Runner.jsonClients(participantParams, envIface).andThen { case Success(ps) =>
+      Future.sequence(
+        partyMap
+          .flatMap { case (party, participant) =>
+            ps.participants
+              .get(participant)
+              .map(_.allocateParty(party, ""))
+          }
+      )
+    }
```

Comment:
It still does not handle Future, as `andThen` does not accept Future. I think flatMap or for-comprehension solves the issue:

```suggestion
    def allocateParties(ps: Participants[JsonLedgerClient]) =
      for {
        (party, participant) <- partyMap
        jsonClient <- ps.participants.get(participant)
      } yield jsonClient.allocateParty(party, "")

    for {
      ps <- Runner.jsonClients(participantParams, envIface)
      _ <- Future.sequence(allocateParties(ps))
    } yield jsonClients
```

Reply:
:+1: